e2e/cypress/integration/notifications/not_unless_Im_mentioned_spec.js

```diff
@@ -0,0 +1,140 @@
+            this.opts = opts;
+        }
+
+        Notification.requestPermission = function() {
+            return 'granted';
+        };
+
+        Notification.close = function() {
+            return true;
+        };
+
+        win.Notification = Notification;
+
+        cy.spy(win, 'Notification').as('notifySpy');
+    });
+
+    // Verify that we now have a Notification property
+    cy.window().should('have.property', 'Notification');
+}
+
+describe('reply-notifications', () => {
+    function ignoreUncaughtException() {
+        cy.on('uncaught:exception', (err) => {
+            expect(err.message).to.include('.close is not a function');
+
+            return false;
+        });
+    }
+
+    let testTeam;
+    let otherChannel;
+    let townsquareChannelId;
+    let receiver;
+    let sender;
+
+    before(() => {
+        // # Update Configs
+        cy.apiUpdateConfig({
+            ServiceSettings: {
+                ExperimentalChannelOrganization: true,
+            },
+        });
+
+        cy.apiInitSetup().then(({team, channel, user}) => {
+            testTeam = team;
+            otherChannel = channel;
+            receiver = user;
+
+            cy.apiCreateUser().then(({user: user1}) => {
+                sender = user1;
+
+                cy.apiAddUserToTeam(testTeam.id, sender.id);
+            });
+
+            cy.apiGetChannelByName(testTeam.name, 'town-square').then((res) => {
+                townsquareChannelId = res.body.id;
+            });
+
+            // # Login as receiver and visit town-square channel
+            cy.apiLogin(receiver);
+            cy.visit(`/${testTeam.name}/channels/town-square`);
+        });
+    });
+
+    it('MM-T551 not unless I\'m mentioned', () => {
+        ignoreUncaughtException();
+
+        // # Set users notification settings
+        setReplyNotificationsSetting('#notificationCommentsNever');
+
+        // # Navigate to town square channel
+        cy.get(`#sidebarItem_${'town-square'}`).scrollIntoView().click({force: true});
+
+        // # post a message
+        cy.postMessage('Hi there, this is a root message');
+
+        // # get post id of message
+        cy.getLastPostId().then((postId) => {
+            // # switch to other channel so that unread notifications in 'town-square` may be triggered
+            cy.get(`#sidebarItem_${otherChannel.name}`).scrollIntoView().click({force: true});
+
+            // # post a message in original thread as another user
+            cy.postMessageAs({sender, message: 'This is a reply to the root post', channelId: townsquareChannelId, rootId: postId});
+        });
+
+        // * Verify stub was not called
+        cy.get('@notifySpy').should('be.not.called');
+
+        // * Verify unread mentions badge does not exist
+        cy.get('#publicChannel').scrollIntoView();
+        cy.get('#sidebarItem_town-square').
+            scrollIntoView().
+            find('#unreadMentions').
+            should('be.not.visible');
```

Comment:
This can simply,
```javascript
cy.get('#sidebarItem_town-square').find('#unreadMentions').should('be.not.visible');
```
or preferably,
```suggestion
        cy.findByLabelText('town square public channel unread').should('be.visible');
```

Reply:
I've applied the first suggestion (dropping `scrollIntoView`).

In this test we're checking the absence of a badge. Although the second suggestion is technically correct, to me it feels unintuitive because we're testing the absence of something by checking the presence of something else.

I think the intent original code is more easy to understand.

Reply:
Thanks for checking and totally makes sense.